File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/KotlinAndroidPlatformConfig.java
package ru.pocketbyte.locolaser.platform.kotlinmobile;

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.KotlinAndroidResources;
import ru.pocketbyte.locolaser.resource.PlatformResources;

public class KotlinAndroidPlatformConfig extends KotlinBaseImplPlatformConfig {

    public static final String TYPE = "kotlin-android";

    private String mAppPackage;

    @Override
    public String getType() {
        return TYPE;
    }

    @Override
    protected String getDefaultResourceName() {
        return "ru.pocketbyte.locolaser.StrAndroid";
    }

    @Override
    public PlatformResources getResources() {
        return new KotlinAndroidResources(getResourcesDir(), getResourceName(), getInterfaceName(), getAppPackage());
    }

    public void setAppPackage(String appPackage) {
        this.mAppPackage = appPackage;
    }

    public String getAppPackage() {
        return mAppPackage;
    }
}

File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/KotlinAndroidPlatformConfig.kt
package ru.pocketbyte.locolaser.platform.kotlinmobile

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.KotlinAndroidResources

class KotlinAndroidPlatformConfig : KotlinBaseImplPlatformConfig() {

    companion object {
        const val TYPE = "kotlin-android"
    }

    override val type =  TYPE

    override val defaultResourcesPath = "./src/androidMain/kotlin/"
    override val defaultResourceName  = "ru.pocketbyte.locolaser.kmpp.AndroidStringRepository"
    override val defaultInterfaceName = "ru.pocketbyte.locolaser.kmpp.StringRepository"

    override val resources
        get() = KotlinAndroidResources(resourcesDir!!, resourceName!!, interfaceName!!)

}